src/main/java/de/tum/in/www1/artemis/domain/Exercise.java

```diff
@@ -478,6 +478,18 @@ public Boolean isEnded() {
         return ZonedDateTime.now().isAfter(getDueDate());
     }
 
+    /**
+     * Checks if the due date is in the future. Returns true, if no due date is set.
+     *
+     * @return true if the due date is in the future, otherwise false
+     */
+    public boolean isBeforeDueDate() {
```

Comment:
Is it intended that this information is sent to the client?
otherwise you need `@JsonIgnore` here